recipes/cuda-nvcc/build.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# Install to conda style directories
+[[ -d lib64 ]] && mv lib64 lib
+
+[[ ${target_platform} == "linux-64" ]] && targetsDir="targets/x86_64-linux"
+[[ ${target_platform} == "linux-ppc64le" ]] && targetsDir="targets/ppc64le-linux"
+[[ ${target_platform} == "linux-aarch64" ]] && targetsDir="targets/sbsa-linux"
+
+for i in `ls`; do
+	[[ $i == "build_env_setup.sh" ]] && continue
+	[[ $i == "conda_build.sh" ]] && continue
+	[[ $i == "metadata_conda_debug.yaml" ]] && continue
+	if [[ $i == "lib" ]] || [[ $i == "include" ]]; then
+		mkdir -p ${PREFIX}/${targetsDir}
+		mkdir -p ${PREFIX}/$i
+		cp -r $i ${PREFIX}/${targetsDir}
+		for j in `ls $i`; do
+			ln -s ../${targetsDir}/$i/$j ${PREFIX}/$i/$j
+		done
+	else
+		cp -r $i ${PREFIX}
+	fi
+done
+
+echo '' >> "${PREFIX}/bin/nvcc.profile"
+echo "NVCC_PREPEND_FLAGS += -ccbin "\""${CXX}"\""" >> "${PREFIX}/bin/nvcc.profile"
```

Comment:
Looks like `$BUILD_PREFIX` is contained in `$CXX`, but is not being expanded. Maybe using [indirect expansion]( https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html ) helps?

```suggestion
echo "NVCC_PREPEND_FLAGS += -ccbin "\""${!CXX}"\""" >> "${PREFIX}/bin/nvcc.profile"
```

Reply:
Seems this results in an empty string. Maybe we can just rely on `BUILD_PREFIX` and simply make replacement more straightforward. Perhaps like so

```suggestion
echo "NVCC_PREPEND_FLAGS += -ccbin "\""${CXX/BUILD_PREFIX/(BUILD_PREFIX)}"\""" >> "${PREFIX}/bin/nvcc.profile"
```

Reply:
After looking at this more, think we need to move this to [activation/deactivation scripts]( https://docs.conda.io/projects/conda-build/en/latest/resources/activate-scripts.html ). The scripts themselves would follow these patterns to [activate]( https://github.com/conda-forge/nvcc-feedstock/blob/9c8e43393d9c798c52a9d8e56785920ef767cabb/recipe/linux/activate.sh#L26-L29 ) and [deactivate]( https://github.com/conda-forge/nvcc-feedstock/blob/9c8e43393d9c798c52a9d8e56785920ef767cabb/recipe/linux/deactivate.sh#L30-L34 ). In particular activate would run

```bash
export NVCC_PREPEND_FLAGS = "${NVCC_PREPEND_FLAGS} -ccbin ${CXX}"
```

Reply:
Why do you need activation scripts? What @adibbley had here was correct.

Reply:
No it didn't actually work. We tried to set this in `nvcc.profile` (also experimented with this a number of different ways locally) and it didn't work. Setting the environment variable in the activation script does work.

Should add personally would have been happier not to use activation scripts (and use `nvcc.profile` or similar), but that doesn't seem to be option.

Reply:
Can you be more specific about what didn't work?

Reply:
It kept trying to use `gcc`, which it could not find instead of the compiler specified here

Reply:
Can you upload the packages to some anaconda.org channel so that I can have a look?

Reply:
Sure. Think it's [this one]( https://anaconda.org/jakirkham/cuda-nvcc/files?version=12.0.76&channel=nvcc_tst )

Reply:
Have you got an x86_64 build?

Reply:
Try 
```bash
echo "NVCC_PREPEND_FLAGS += -ccbin "\"${PREFIX}/bin/$(basename ${CXX})\""">> "${PREFIX}/bin/nvcc.profile"
```

Reply:
No sorry. Only the ARM build

Yeah we tried a bunch of stuff like this without luck. AFAICT `NVCC_PREPEND_FLAGS` isn't respected in `nvcc.profile`

Reply:
If we soft/hard linked `$PREFIX/bin/gcc` to `$CXX`, that would work (for example the `compilers` package works), but that comes with its own set of tradeoffs

Reply:
`nvcc.profile` does seem to support `compiler-bindir` (same as `--compiler-bindir` CLI option), but that only specifies what directory to look for `gcc` in (not the name of `gcc`)

Reply:
`--compiler-bindir` and `-ccbin` are aliases right?

Reply:
Yeah I tried setting `compiler-bindir` to the exact path of `$CXX` in `nvcc.profile`, but got the same error. The only examples I've seen of using this in `nvcc.profile` have been on Windows. It might be ignored on Linux.

Reply:
We can add a symlink in the compiler package like `${PREFIX}/bin/x86_64-conda-linux-gnu-gcc -> ${PREFIX}/x86_64-conda-linux-gnu/bin/gcc`

Reply:
That seems like an interesting option. Would that be added to the `$PATH` then?

Reply:
No, you can use `${PREFIX}/x86_64-conda-linux-gnu/bin/` as `--compiler-bindir` then.

Reply:
Ok not following how this would be different here. Given `nvcc.profile` already isn't respecting `compiler-bindir` on Linux, this flag would still be handled via an environment variable in an activation script as it is now. What am I missing?

Reply:
nvcc.profile does support `compiler-bindir`, just not the compiler name. i.e. it supports a custom directory name, but not a custom compiler name. So having `gcc` in `${PREFIX}/x86_64-conda-linux-gnu/bin/` would help.

Reply:
It supports both. From [the docs]( https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#compiler-bindir-directory-ccbin ):

> Specify the directory in which the default host compiler executable resides.
>
> The host compiler executable name can be also specified to ensure that the correct host compiler is selected.

Reply:
In any event `nvcc.profile` still seems to be ignoring `compiler-bindir` on Linux

Given we don't seem to be coming up with a better option than the activation script, would suggest we stick with it since we know it works

Reply:
Try the following
```
mamba create -n nvcc cuda-nvcc cuda-cudart-dev cuda-cudart-static compilers -c nvidia
mamba activate nvcc
mv $CONDA_PREFIX/bin/nvcc.profile $CONDA_PREFIX/bin/nvcc.profile.bak
cp $CONDA_PREFIX/bin/nvcc.profile.bak $CONDA_PREFIX/bin/nvcc.profile
echo "compiler-bindir = $CONDA_PREFIX/$HOST/bin" >> $CONDA_PREFIX/bin/nvcc.profile
ln -sf $CONDA_PREFIX/bin/$HOST-gcc $CONDA_PREFIX/$HOST/bin/gcc
ln -sf $CONDA_PREFIX/bin/$HOST-g++ $CONDA_PREFIX/$HOST/bin/g++
echo "int main() {}" > test.cu
nvcc test.cu --verbose
```